### PR TITLE
Add NimScript support to Nimble

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: src/nimble.nim [hunk 1]
```diff
@@ -112,21 +117,21 @@ proc writeVersion() =
       [nimbleVersion, CompileDate, CompileTime])
   quit(QuitSuccess)
 
-proc getNimbleDir(options: Options): string =
-  options.config.nimbleDir
+proc getNimbleDir(opt: Options): string =
```
Comment (dom96) [inline src/nimble.nim]: why?

Comment (Araq) [inline src/nimble.nim]: Because I found it slightly confusing with the compiler/options import.

**Comment by andreaferretti** [comment]
Is this being discussed somewhere? It would be great if this was merged, so that one could use a single `config.nims` to write tasks and dependencies!

**Comment by andreaferretti** [comment]
Are there any news about this PR?

**Comment by dom96** [comment]
Yeah. I'm working on it. Needs some adjustments.

**Comment by dom96** [comment]
Note to self: ``WARNING: No .nimble file found for /Users/dom/.nimble/pkgs/jester-1.0.0`` need to fix this if installed with a nimscript.

**Comment by dom96** [comment]
Current blockers:
  * The bug above.
  * ``setCommand "foo"``
  * The ``.nimble`` extension should be reused for nimscript files, instead of using ``.nims``.

**Comment by andreaferretti** [comment]
Thaank you, nice Christmas present! ;-)

**Comment by dom96** [comment]
No problem. Please test it if you can :)

There are still some issues to work out of course. Some which I am aware of, and probably many others which I am not.

**Pull request merged**